hoodie-client/src/main/java/com/uber/hoodie/io/HoodieCommitArchiveLog.java

```diff
@@ -58,44 +71,84 @@ public HoodieCommitArchiveLog(HoodieWriteConfig config, FileSystem fs) {
             .getArchiveLogPath(config.getBasePath() + "/" + HoodieTableMetaClient.METAFOLDER_NAME);
     }
 
+    private HoodieLogFormat.Writer openWriter() {
+        try {
+            if(this.writer == null) {
+                return HoodieLogFormat.newWriterBuilder()
+                        .onParentPath(archiveFilePath.getParent())
+                        .withFileId(archiveFilePath.getName())
+                        .withFileExtension(HoodieLogFile.DELTA_EXTENSION)
+                        .withFs(fs)
+                        .overBaseCommit("").build();
+            } else {
+                return this.writer;
+            }
+        } catch(InterruptedException | IOException e) {
+            throw new HoodieException("Unable to initialize HoodieLogFormat writer", e);
+        }
+    }
+
+    private void close() {
+        try {
+            if(this.writer != null) {
+                this.writer.close();
+            }
+        } catch(IOException e) {
+            throw new HoodieException("Unable to close HoodieLogFormat writer", e);
+        }
+    }
+
     /**
      * Check if commits need to be archived. If yes, archive commits.
      */
     public boolean archiveIfRequired() {
-        List<HoodieInstant> commitsToArchive = getCommitsToArchive().collect(Collectors.toList());
-        boolean success = true;
-        if (commitsToArchive.iterator().hasNext()) {
-            log.info("Archiving commits " + commitsToArchive);
-            archive(commitsToArchive);
-            success = deleteInstants(commitsToArchive);
-        } else {
-            log.info("No Commits to archive");
+        try {
+            List<HoodieInstant> commitsToArchive = getCommitsToArchive().collect(Collectors.toList());
+            boolean success = true;
+            if (commitsToArchive.iterator().hasNext()) {
+                this.writer = openWriter();
+                log.info("Archiving commits " + commitsToArchive);
+                archive(commitsToArchive);
+                success = deleteInstants(commitsToArchive);
+            } else {
+                log.info("No Commits to archive");
+            }
+            return success & deleteOtherInstants();
+        } catch(Exception e) {
+            throw new HoodieException("Unable to archive commits", e);
+        } finally {
+            close();
         }
-        return success & deleteOtherInstants();
     }
 
     private boolean deleteOtherInstants() {
         // Delete clean and rollback files
-        List<HoodieInstant> toDelete = getInstantsToDelete().collect(Collectors.toList());
+        List<HoodieInstant> toDelete = getOtherInstantsToDelete().collect(Collectors.toList());
         if(!toDelete.isEmpty()) {
+            this.writer = openWriter();
+            archive(toDelete);
```

Comment:
you are not deleting any more. This should be merged with getCommitstoArchive (rename this as getInstantsToArchive) and archive - Just one way to collect everything that needs to be archived and archive it. You dont have to treat commit as something special. Also make sure the method names reflect what you do in the methd.